shark_turbine/kernel/wave/register_analysis.py

```diff
@@ -33,16 +42,43 @@ def set_register_shape(trace: CapturedTrace, custom: CustomOp) -> None:
                     idx
                 ]
             )
-            set_register_shape(trace, iter_arg)
+            set_register_shape(trace, iter_arg, vector_map)
             custom.fx_node.thread_shape = iter_arg.fx_node.thread_shape
             break
+        elif isinstance(custom_user, ReduceOp):
+            get_thread_shape = lambda reg_shape: max(reg_shape)
+            # Check that dim is non-reduction && in hw_constraint.vector_shape.
+            is_valid_dim = lambda dim: dim != custom_user.dim and dim in vector_map
```

Comment:
If vector_map is None (if no vector shapes are specified), will this fail?

Reply:
If vector_map is None, it will fail in the assertion on below checking for that rank(thread_shape) = rank(symbolic_shape) - num_reduction_dims.

Reply:
do you want a earlier/more explicit assert for vector_map != None?

Reply:
No just wanted to make sure we would catch the failure case which sounds like it does, thanks!